### ISSUE TITLE
compile wasi version wasm file

### ISSUE THREAD

**Issue by mccoysc** [pyodide/pyodide#3535]
## 🚀 wasi support
as we know,it's not easy to sandbox python,but wasi can.

and in nodejs,wasi interface can be used as a sandboxed system call interface.
would you give a wasi version pyodide?

**Comment by mccoysc** [comment]
with wasi supported,pyodide can have a little size js helper code  in nodejs(or can run in web broswer with the help of wasmer).
and can run in wasmtime(or other wasm runtime) with any js helper code.

**Comment by mccoysc** [comment]
the little helper code for webbrowser:
![image](https://user-images.githubusercontent.com/18025276/216290351-936fd10b-8b50-4760-80b0-94710bf1129d.png)

but now pyodide js helper code is more than 1.6MB .

**Comment by mccoysc** [comment]
@hoodmane

**Comment by ryanking13** [comment]
I don't think we will support WASI in any time. We are relying on Emscripten-specific features too much. Closing as not planned.
